narayana/ArjunaJTA/jta/pom.xml

```diff
@@ -104,6 +104,9 @@
                                 <transformer implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
                                     <mainClass>org.openjdk.jmh.Main</mainClass>
                                 </transformer>
+                                <transformer implementation="org.apache.maven.plugins.shade.resource.AppendingTransformer">
```

Comment:
@tomjenkinson this transformer together with the removal of some dependencies (resteasy-core, undertow...) solved the error I encountered: "java.lang.ClassNotFoundException: org.jboss.jbossts.star.provider.HttpResponseException"
The transformer is needed by the maven-shade-plugin otherwise a resteasy error is thrown: "jakarta.ws.rs.ProcessingException: RESTEASY003215: could not find writer for content-type application/x-www-form-urlencoded type: jakarta.ws.rs.core.Form"

Reply:
Thank you for the update! So is org.jboss.jbossts.star.provider.HttpResponseException in the Jar now then? Seems like finding a solution would have been tricky to spot. I guess there are a couple of dependencies that include a META-INF/services/jakarta.ws.rs.ext.Providers then? And these are being merged together for the uber Jar (if I understand https://maven.apache.org/plugins/maven-shade-plugin/examples/resource-transformers.html#AppendingTransformer correctly)?

Reply:
@marcosgopen what was the outcome of this conversation?

Reply:
Yes, now the command 'jar tf target/benchmarks.jar | grep HttpResponseException' confirms that org.jboss.jbossts.star.provider.HttpResponseException is in the jar. You are right about the AppenderTransformer: using the transformer seems the only way to avoid overwriting our needed dependency.

Reply:
@mmusgrov after the migration the new dependencies need this appendingTransformer in order to avoid a overwriting during the creation of the benchmark jar. Otherwise we encounter the runtime error described above. Sorry, I forgot to answer Tom's questions before.

Reply:
Normally we would need to wait for the reviewer (ie @tomjenkinson ) to mark the conversation as resolved if he is happy with the response but I think you have answered the question.